JustSaying/AwsTools/MessageHandling/SnsTopicBase.cs

```diff
@@ -72,9 +73,10 @@ public async Task PublishAsync(Message message, PublishMetadata metadata, Cancel
             catch (Exception ex)
             {
                 if (!ClientExceptionHandler(ex, message))
+                {
                     throw new PublishException(
-                        $"Failed to publish message to SNS. TopicArn: {request.TopicArn} Subject: {request.Subject} Message: {request.Message}",
-                        ex);
+                        $"Failed to publish message to SNS. TopicArn: {request.TopicArn} Subject: {request.Subject} Message: {request.Message}", ex);
```

Comment:
Should have a separator (semi-colon?) between the elements in the second sentence.

```
Topic ARN: {request.TopicArn}; Subject: {request.Subject}; Message: {request.Message}.
```